internal/server/subscription_manager.go

```diff
@@ -205,3 +211,20 @@ func (c *subscriptionManagerCore) unsubscribe(name string) {
 		delete(c.subscriptions, name)
 	}
 }
+
+// cleanSubscriptionsAndEstimateSize clones the given slice and removes duplicate elements by sorting
+// it. This ensures that the server does not process the same subscription twice for the same
+// request. It then estimates the size of send buffer to pass to
+// [BatchSubscriptionHandler.StartNotificationBatch] if a [SendBufferSizeEstimator] was provided.
+func (c *subscriptionManagerCore) cleanSubscriptionsAndEstimateSize(
+	resourceNamesSubscribe []string,
+	initialResourceVersions map[string]string,
+) (cleaned []string, size int) {
+	cleaned = slices.Clone(resourceNamesSubscribe)
+	slices.Sort(cleaned)
+	cleaned = slices.Compact(cleaned)
+	if len(initialResourceVersions) > 0 && c.sizeEstimator != nil {
```

Comment:
shouldn't this be `len(initialResourceVersions) == 0`? we only want to use sizeEstimator when IRV is not provided, right?

Reply:
In fact, why not always estimate the size normally, and check whether initialResourceVersions is empty directly in StartNotificationBatch. Seems more straightforward to put the check right before the buffer swapping.